### PR TITLE
Change hash function in GroupRandomly 

### PR DESCRIPTION
**Pull request by MansurAshraf**
Current hashing is causing collision so switching to identity hash.

### CODE REVIEW

**Comment by MansurAshraf** [comment]
Working on a unit test....

**Comment by jnievelt** [comment]
This goes better in the TypedPipe object?

**Comment by MansurAshraf** [comment]
I guess I can move it there and make it private. I inlined it because I dont want anyone using it outside this method

**Comment by Gabriel439** [comment]
Looks good to me

File: scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala [hunk 1]
```diff
@@ -382,7 +386,17 @@ trait TypedPipe[+T] extends Serializable {
   def groupRandomly(partitions: Int): Grouped[Int, T] = {
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala]: can we use this for `.shard` now?

**Comment by ianoc** [comment]
LGTM, thanks for getting on this quickly

**Pull request merged**